### DISCUSSION TITLE
ERROR: LoadError: AssertionError: Target platform is Windows, but can’t find valid FMU binary

### DISCUSSION THREAD

**Discussion by Amuri44** [ThummeTo/FMI.jl#81] [Q&A] [answered]
I am using Julia v. 1.6.3 64bit in my windows machine. I have an FMU model (generated by 32bits simulation tool) which I tried to load it. However, I got the below error.
```julia
ERROR: LoadError: AssertionError: Target platform is Windows, but can't find valid FMU binary at `` for path `path/to/file/myFMU`.
```
I am wondering if the problem is because:
1- there are some other missing files that need to be existed? or
2- this julia is 64bit and the FMU model is 32bit. If so, how can I let julia reads the 32bits FMU model? do I need to uninstall the current version and then install 32bit julia or there is a better way?
This is what I get when from REPL of VS Code:
```julia
julia> versioninfo()
Julia Version 1.6.3
Commit ae8452a9e0 (2021-09-23 17:34 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, skylake)
Environment:
  JULIA_EDITOR = code
  JULIA_NUM_THREADS = 6
```

**Comment by ThummeTo** [comment, marked as answer]
Dear @Amuri44,

a 64-bit application is not capable of loading (and executing) a 32-bit binary. Because you are using a 64-Bit application, there are no executable files in a 32-bit FMU. So if you want to use a 32-bit FMU, you can try to install a 32-bit Julia and run it in compatibility mode on your 64-bit OS. See here: [Stackoverflow](https://stackoverflow.com/questions/2265023/load-32bit-dll-library-in-64bit-application).
And finally, I did not implement 32-bit support, but if someone is using it and Julia is availabel for 32-bit Windows I can do so (only 3 lines of code).

**EDIT: 32-bit support added in version 0.3.11**

Best regards,
Tobias

PS: This is more a question. For Q&A, please use the discussion function. The "issue"-function is meant to be for library related issues/bugs. I will move this topic to discussions.

**Reply by ThummeTo** [reply]
EDIT: I will add support for 32-bit FMUs, but because all our testing-FMUs are 64-bit, we will (for now) only automatically test for 64-bit operation systems. Independent of that, you need to use 32-bit Julia.

**Reply by ThummeTo** [reply]
I added 32-bit support for FMI.jl 0.3.11 (coming in around an hour). But this is not tested, I think some of the dependencies might not be compatible with 32-bit applications.

**Reply by Amuri44** [reply]
Dear @ThummeTo ,
Thank you very much for your explanations and your update of `FMI.jl`.
I have downloaded julia 32-bit on my (64-bit OS) and downloaded the `FMI.jl 0.3.11`. It seems to work for me at this moment.

**Reply by ThummeTo** [reply]
Good to hear! Let me know if there are issues!

**Reply by Amuri44** [reply]
Sure, I will keep you updated. Thanks for your work.
